module/VuFind/src/VuFind/View/Helper/Root/MakeLink.php

```diff
@@ -0,0 +1,109 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+
+/**
+ * Make link view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class MakeLink extends AbstractHelper
+{
+    /**
+     * Create an anchor tag
+     *
+     * $href will override $attrs['href']
+     * > Feel free to use like makeLink('text', 'href', $defaults);
+     *
+     * If no $href, will try to find an href in $attrs
+     * > makeLink('text', null, ['href' => '#', 'class' => 'btn-link'])
+     *
+     * If $attrs is a string, will be treated like a class
+     * > makeLink('text', $href, 'btn-link')
+     *
+     * @param string       $text  Link contents (should be properly-formed HTML)
+     * @param string|array $href  Link destination (null to skip)
+     * @param array        $attrs Link attributes (associative array)
+     *
+     * @return string HTML for an anchor tag
+     */
+    public function __invoke($text, $href = null, $attrs = [])
+    {
+        // $attr not an object, interpret as class name
+        if (!is_array($attrs)) {
+            $attrs = !empty($attrs) ? ['class' => $attrs] : [];
+        }
+
+        // Merge all attributes
+        $mergedAttrs = array_merge(
+            $attrs ?? [],
+            !empty($href) ? ['href' => $href] : []
+        );
+
+        // just text
+        if (empty($mergedAttrs)) {
+            return $text;
+        }
+
+        // Span instead of anchor when no href present
+        if (!($mergedAttrs['href'] ?? false)) {
+            return $this->compileAttrs($text, $mergedAttrs, 'span');
+        }
+
+        // Compile attributes
+        return $this->compileAttrs($text, $mergedAttrs);
+    }
+
+    /**
+     * Turn associative array into a string of attributes in an anchor
+     *
+     * @param string $text    Link contents
+     * @param array  $attrs   Link attributes (associative array)
+     * @param string $tagName HTML tag name
+     *
+     * @return string
+     */
+    protected function compileAttrs($text, $attrs, $tagName = 'a')
```

Comment:
I find the name `compileAttrs` a bit misleading since it actually does most of the real work and compiles the tag, it's content etc. as well. Something like `compileTag` might be more appreciative of everything this does, but compileAttrs could still be a method that strings the attributes together.

Reply:
`$text` => `$innerHtml` or something to indicate it's HTML, not just text?

Reply:
I had that very thought as I was driving to work this morning -- $innerHtml is a much more meaningful variable name here.